test/python/transpiler/test_collect_2q_blocks.py

```diff
@@ -102,51 +102,11 @@ def test_block_interrupted_by_gate(self):
 
         self.assertEqual(dag_nodes, pass_nodes)
 
-    def test_block_with_classical_register(self):
-        """Test that only blocks that share quantum wires are added to the block.
-        It was the case that gates which shared a classical wire could be added to
-        the same block, despite not sharing the same qubits. This was fixed in #2956.
-
-                                    ┌─────────────────────┐
-        q_0: |0>────────────────────┤ U2(0.25*pi,0.25*pi) ├
-                     ┌─────────────┐└──────────┬──────────┘
-        q_1: |0>──■──┤ U1(0.25*pi) ├───────────┼───────────
-                ┌─┴─┐└──────┬──────┘           │
-        q_2: |0>┤ X ├───────┼──────────────────┼───────────
-                └───┘    ┌──┴──┐            ┌──┴──┐
-        c0_0: 0 ═════════╡ = 0 ╞════════════╡ = 0 ╞════════
-                         └─────┘            └─────┘
-
-        Previously the blocks collected were : [['cx', 'u1', 'u2']]
```

Comment:
why does this test need to be removed?

Reply:
This test is confusing and useless. Maybe at some point this pass was able to handle classical conditions, this is not the case anymore. Gates with classical conditions are not collected. (You can see that on the test that comes afterwards)